### PR TITLE
Enable Collection Owner feature

### PR DESCRIPTION
**Pull request by JohnChangUK**
## Description
As part of [AIP 95 collections permissions update](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-95.md), enable feature Collection Owner. This enables users to create collections as the collection owner. 

## How Has This Been Tested?
New functions added and tested.

## Type of Change
- [X] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [X] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Move Compiler
- [ ] Other (specify)

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: aptos-move/aptos-release-builder/data/collection-owner.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          git_hash: ~
+  - name: gas
+    metadata:
+      title: ""
+      description: ""
+      source_code_url: ""
+      discussion_url: ""
+    execution_mode: MultiStep
+    update_sequence:
+      - DefaultGas
```
Comment (sherry-x) [inline aptos-move/aptos-release-builder/data/collection-owner.yaml:23]: you don't need anything above this line

File: aptos-move/aptos-release-builder/data/collection-owner.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      title: ""
+      description: ""
+      source_code_url: ""
+      discussion_url: ""
+    execution_mode: MultiStep
+    update_sequence:
+      - FeatureFlag:
+          enabled:
+            - collection_owner
+          disabled: []
```
Comment (sherry-x) [inline aptos-move/aptos-release-builder/data/collection-owner.yaml:35]: you don't need "disabled" either

Comment (sherry-x) [reply aptos-move/aptos-release-builder/data/collection-owner.yaml:35]: also, target this PR on 1.23 branch.

**Comment by JohnChangUK** [comment]
Updated PR: https://github.com/aptos-labs/aptos-core/pull/15420

**Pull request closed without merging**